### PR TITLE
Add mac hash to user agent header

### PR DESCRIPTION
**Pull request by balajikris**
The content to be added to Changelog is as follows:
* add a SHA-256 hash of macAddress to userAgent header, per VS Telemetry standard.

### CODE REVIEW

File: lib/util/telemetry.js [hunk 1]
```diff
@@ -297,7 +303,63 @@ function constructUserAgentData() {
...
   }
 }
+
+
+var _getMacAddressHash = function () {
+  var networkInterfaces = os.networkInterfaces();
+  var interfaceKeys = Object.keys(networkInterfaces);
+
+  var nicPattern;
+  if (os.platform().startsWith('win')) {
```
Comment (carlhoerberg) [inline lib/util/telemetry.js]: `os.platform()` does not return a string, it's an object!

Comment (balajikris) [inline lib/util/telemetry.js]: Not sure where you're seeing its an `object`.

But looking at https://nodejs.org/api/os.html#os_os_platform
~~~
os.platform()
Added in: v0.5.0
Returns: <string>

Currently possible values are:
'aix'
'darwin'
'freebsd'
'linux'
'openbsd'
'sunos'
'win32'
~~~

**Comment by carlhoerberg** [comment]
It's making azure-cli unusable, crashes at every command. See issue #3546

**Comment by balajikris** [comment]
yep, noticed the bug after i responded to this comment. I've responded to the bug now. Check it out and let me know if it worked.

**Pull request merged**